### ISSUE TITLE
Child process will not exit on Windows

### ISSUE THREAD

**Issue by caspervonb** [caspervonb/amok#153]
The test suite is currently stalling on windows. They pass but the process doesn't terminate.
I know it has *something* to do with child_process.spawn basically never closing the child. This seems to only occur for tests that call the `compiler` function via the API.

**Comment by caspervonb** [comment]
So basically, it turns out on Windows. This will not kill the process.

```js
var child_process = require('child_process');

var ps = child_process.spawn('.\\node_modules\\.bin\\babel.cmd', ['cp.js', '--out-file', 'output.js', '--watch']);
setTimeout(function () {
  ps.kill();
}, 2000);
```

**Comment by caspervonb** [comment]
One workaround is to close the child process's streams with either `destroy` or `end`.

Another is to just use node.exe directly, it works fine if there's no shell wrapper. The latter would have to be fixed in `compiler_process`

**Comment by stereokai** [comment]
Is there anything wrong with the alternatives you're suggesting?

**Comment by caspervonb** [comment]
> One workaround is to close the child process's streams with either destroy or end.

This diddn't actually work as the child process would be running detatched. Sure the parent exits but this leads to an ever bigger bug with old compilers still compiling.

> Another is to just use node.exe directly, it works fine if there's no shell wrapper. The latter would have to be fixed in compiler_process

This is just not how npm executables work, and feels like a major hack for something I'd imagine is a known and solved issue.

**Comment by caspervonb** [comment]
Alright, we'll go with the second option. Skip the shell wrappers and spawn node directly. We can resolve the package and find the relevant bin path from there.

**Comment by caspervonb** [comment]
> Alright, we'll go with the second option. Skip the shell wrappers and spawn node directly. We can resolve the package and find the relevant bin path from there.

So I did that but am not happy with the semantic differences it imposes. There's value in following expectations,  the assumption that the compiler in your `PATH` is the one that will be used should hold true.

**Comment by caspervonb** [comment]
We can resolve the link in `.cmd` files, makes it semantically still the same. Skipping the extra shell.

```js
which(name, function(error, script) {
  if (path.extname(command).toLowerCase() === '.cmd') {
    var content = fs.readFileSync(command);
    var matches = content.match(/node  "%~dp0\\(.*?)"/);

    script = matches[1];
  } 

  var ps = child.fork(script);
});
```

**Comment by code-guru** [comment]
I am not sure if the issue I am going discuss is related to this one or not.

Here is my issue; Amok does not work in windows when amok itself opens the chrome window. I am on win7 with `iojs 2.1.0` and `amok 1.1.3`.

I try to investigate the problem, and I found out that when amok opens chrome window, it does not connect to chrome remote debug, and it waits. As soon as I close the chrome window, amok try to connect to the debug port. 

It seems somewhere in the amok code, it does not continue the process after spawning a chrome window.

I use the following command:
`amok.cmd --browser chrome --debug-port 4000 http://localhost:3000`

**Comment by caspervonb** [comment]
Resolved in `compiler_process@1.0.1`. Pushed it as a patch since it wont break anything, even tho its a behavioural change i consider it a bug.

@code-guru Unrelated issue, this is about the compiler process not closing after amok has been closed on windows due to how processes work on windows compared to unix. It does not happen with executables only node scripts which have an intermediary cmd.exe shell. Killing the process kills the intermediary cmd.exe but not the node.exe process itself which has the compiler script.

I your issue would be #94, which I should've fixed a long time ago but I need to break the API. Which I am doing for 2.0.

Basically the issue is that we're checking with strict equality for the tab to be ready and available before moving on from the initialiser that spawns the browser.

There's supposed to be a timeout in there for two minutes to at-least give an error about the issue but that was broken too (fixed in f43bf4776c1c79ee52272a15b232a70311127c05) but I'm heading towards a 2.0 release.

Anyhow, If I am right about what the issue is, the fix is simple.

```sh
amok.cmd --browser chrome --debug-port 4000 http://localhost:3000/
```

Note the trailing slash, since `'http://localhost:4000' !== 'http:localhost:4000/'` it will get stuck.

If I am wrong about the issue, please open a new issue with the information you get when running amok with the `NODE_DEBUG` env variable set to `amok-browser`.
